src/cache.rs

```diff
@@ -107,17 +107,20 @@ pub struct Cache {
 }
 
 impl Cache {
-    pub fn new<P: AsRef<Path>>(
+    pub fn new(
         name: &'static str,
-        cache_dir: Option<P>,
+        cache_dir: Option<PathBuf>,
         cache_config: CacheConfig,
-    ) -> Self {
-        Cache {
+    ) -> io::Result<Self> {
```

Comment:
Just wondering what people think of this. I know there is even a clippy lint enforcing that `new` only returns `Self`.

Reply:
Right, this should be renamed to `create` or something similar. The lint was backed out at some point, but it's still a good habit.

Reply:
Oh, interesting. I've never spotted any consistency in this.  Though glancing through the stdlib the only example I can find of `::new()` returning `Result` is `CString` right now.  (I know that stdlibs are rarely an example of how to do things the right way after a few years...).

> `create` or something similar.

"or something similar"?  Would be good to have a reasonable convention here if we don't like new returning result.

> The lint was backed out at some point, but it's still a good habit.

Let's re-add it if we choose to prefer this.

Reply:
https://rust-lang.github.io/rust-clippy/master/index.html#new_ret_no_self

Reply:
That does seem to allow `Result<Self>`

Reply:
I would not expect `new` to return result. `CString` is an outlier to me.

Reply:
Now with `s/new/from_config/`